tests/subsys/edac/ibecc/src/ibecc.c

```diff
@@ -336,15 +334,16 @@ void test_main(void)
 		k_oops();
 	}
 #endif
+	return NULL;
+}
 
-	ztest_test_suite(ibecc,
-			 ztest_unit_test(test_trigger_nmi_handler),
-			 ztest_unit_test(test_ibecc_initialized),
-			 ztest_unit_test(test_ibecc_api),
-			 ztest_unit_test(test_edac_dummy_api),
-			 ztest_unit_test(test_ibecc_error_inject_api),
-			 ztest_unit_test(test_ibecc_error_inject_test_cor),
-			 ztest_unit_test(test_ibecc_error_inject_test_uc)
-			);
-	ztest_run_test_suite(ibecc);
+ZTEST(ibecc, test_ibecc_injection) {
+	test_trigger_nmi_handler();
```

Comment:
Are you moving them in one test because test shuffle may break it?

Reply:
:+1: this seems odd, can these not be tested individually? I recall a few days ago they were mostly separate tests instead of just 1.

Reply:
test_ibecc_api_before_inject_test(); must stay before any injection function of test_ibecc_error_inject_api(); test_ibecc_error_inject_test_cor(); test_ibecc_error_inject_test_uc();

Because after injection, some items checking below can not pass: 
ret = edac_ecc_error_log_get(dev, &value);
zassert_equal(ret, -ENODATA, "edac_ecc_error_log_get failed");

ret = edac_errors_cor_get(dev);
zassert_equal(ret, 0, "Error correctable count not zero");

ret = edac_errors_uc_get(dev);
zassert_equal(ret, 0, "Error uncorrectable count not zero");

ret = edac_parity_error_log_get(dev, &value);
zassert_equal(ret, -ENODATA, "edac_parity_error_log_get failed");

when shuttle enabling, there are 3 times default running tests. we can not ensuring the sequence if we separate each testing.

Reply:
in order to pass the shuttle, I will modify the testing context itself. will update one.

Reply:
> +1 this seems odd, can these not be tested individually? I recall a few days ago they were mostly separate tests instead of just 1.

There were separate tests, but the order does matter. At least it does make sense to keep API tests before error injection tests. Injection test is changing memory to trigger memory error detection, so the system may not be stable after that.

Reply:
@finikorg That's valid if the system will be in a bad state, I'm OK with this change, but I would use recommend using the suite's predicate in order to make sure that one unstable test runs last (or split it into a different binary using twister.